scripts/check_format_cpp.sh

```diff
@@ -41,6 +41,17 @@ done
 # stop right here if clang-format does not exist in $PATH
 command -v $clang_format_executable >/dev/null 2>&1 || { echo >&2 "clang-format executable '$clang_format_executable' not found.  Aborting."; exit 1; }
 
+# check for the correct clang-format version
+clang_format_version="$(${clang_format_executable} --version)"
+clang_format_major_version=$(echo "${clang_format_version}" | sed 's/^[^0-9]*\([0-9]*\).*$/\1/g')
+clang_format_minor_version=$(echo "${clang_format_version}" | sed 's/^[^0-9]*[0-9]*\.\([0-9]*\).*$/\1/g')
+clang_format_patch_version=$(echo "${clang_format_version}" | sed 's/^[^0-9]*[0-9]*\.[0-9]*\.\([0-9]*\).*$/\1/g')
+if [ "${clang_format_major_version}" -ne 14 ] || [ "${clang_format_minor_version}" -ne 0 ] || [ "${clang_format_patch_version}" -ne 0 ]; then
+  echo "***   ArborX requires clang-format version 14.0.0,"
+  echo "***   but version ${clang_format_major_version}.${clang_format_minor_version}.${clang_format_patch_version} was found instead."
```

Comment:
I still believe that this is a cleaner way to achieve the purpose:
```suggestion
clang_format_required_version="14.0.0"
clang_format_version="$(${clang_format_executable} --version | awk -F '[ -]' '{print $3}')"
if [ "${clang_format_major_version}" -ne "${clang_format_required_version}" ]; then
  echo "***   ArborX requires clang-format version ${clang_format_required_version},"
  echo "***   but version ${clang_format_version} was found instead."
```

Reply:
I think the regex version is still more robust but won't go against a majority, of course.

Reply:
> I think the regex version is still more robust but won't go against a majority, of course.

Why do you think that?

Reply:
The regex tries to find the first appearance of an "x.y.z" type string anywhere in the output. The `awk` version relies on knowing the exact position of this pattern.

Reply:
Hmm. While it is highly unlikely that the output will change, I see your point.